site/_posts/2020-06-05-private-labeling-with-fusionauth.md

```diff
@@ -0,0 +1,107 @@
+
+You can change the token issuer, password rules, and many other settings at the tenant level. The look and feel of the login, forgot password and all other OAuth pages can be customized per tenant. You can also duplicate an existing tenant to easily start from a solid set of defaults. 
+
+When you have two or more tenants, the admin UI displays a new column showing you with which tenant each application is associated.
+
+{% include _image.liquid src="/assets/img/blogs/private-label-with-tenants/applications-with-multiple-tenants.png" alt="Applications when there are multiple tenants." class="img-fluid" figure=false %}
+
+Normally a user's email address is unique across FusionAuth. But each tenant is a separate userspace, so you can have two different user accounts with the same email, but different data, passwords and application associations.
+
+{% include _image.liquid src="/assets/img/blogs/private-label-with-tenants/duplicate-example-users.png" alt="Two users with the same email address." class="img-fluid" figure=false %}
+
+For administrators, there are significant benefits with tenants. You get ample separation as mentioned above. But as an admin, you have one view into all system activity. You also only have one FusionAuth installation to manage, secure, and upgrade. 
+
+Operations has one place to go to set up new API keys or webhooks. If your customer service reps need to reset a password, they don't have to track down the correct FusionAuth installation. Central user management makes their lives easier.
+
+## How to create tenants via API
+
+Even better, tenant creation can be automated. Every time TodoInc sells a premium subscription, code can automatically create a corresponding tenant, like so:
+
+```ruby
+def create_new_tenant(generic_tenant_client, name)
+  # create a new tenant based on the default tenant
+  tenants_response = generic_tenant_client.retrieve_tenants
+  if tenants_response.status != 200
+    puts "Unable to retrieve tenants."
+    return
+  end
+
+  default_tenant_id = tenants_response.success_response.tenants.select { |t| t.name == 'Default' }[0].id
+  default_tenant_theme_id = tenants_response.success_response.tenants.select { |t| t.name == 'Default' }[0].themeId
+
+  new_tenant_request = { "sourceTenantId": default_tenant_id, tenant: {"name": "New client - "+name }}
+  new_tenant_response = generic_tenant_client.create_tenant(nil, new_tenant_request)
+
+  if new_tenant_response.status != 200
+    puts "Unable to create tenant."
+    puts new_tenant_response.error_response
+    return false
+  end
+  new_tenant = new_tenant_response.success_response.tenant
+  [default_tenant_theme_id, new_tenant]
+end
+```
+
+In this code, we find the tenant with the name `Default` and duplicate it, creating a new tenant with the same settings. We also create a new theme for the new tenant. 
```

Comment:
I don't see in the code that it creates a new theme. It looks like it returns a tuple of the default theme id and the new tenant object though.

Reply:
Whoops, thanks.